### ISSUE TITLE
expect transformCredentialInput to move top-level props of W3C credentials inside vc

### ISSUE THREAD

**Issue by joshbax189** [decentralized-identity/did-jwt-vc#73]
Based on my reading of the W3C spec, it seems as though top-level optional props like `evidence` and `credentialSubject` should be copied inside the `vc` claim in the resulting JWT encoded credential. 

However, when you pass `transformCredentialInput` an arg of type `CredentialPayload` they are copied to the top level of the JWT:

```
const cred = {
  '@context': ['https://www.w3.org/2018/credentials/v1'],
  credentialSubject: {
    id: 'did:ethr:0x435df3eda57154cf8cf7926079881f2912f54db4',
    bar: 'baz',
  },
  evidence: { foo: 'bar' },
  type: ['VerifiableCredential'],
};

transformCredentialInput(cred);
```
Results in
```
{
  vc: {
    credentialSubject: { bar: 'baz' },
    '@context': [ 'https://www.w3.org/2018/credentials/v1' ],
    type: [ 'VerifiableCredential' ]
  },
  evidence: { foo: 'bar' },
  sub: 'did:ethr:0x435df3eda57154cf8cf7926079881f2912f54db4'
}
```
Expect
```
{
  vc: {
    credentialSubject: { bar: 'baz' },
    '@context': [ 'https://www.w3.org/2018/credentials/v1' ],
    type: [ 'VerifiableCredential' ],
    evidence: { foo: 'bar' }
  },
  sub: 'did:ethr:0x435df3eda57154cf8cf7926079881f2912f54db4'
}
```
A workaround is to move the props inside a manually constructed `vc` object. But then it's unclear whether all of the props should be manually transformed to match the `JwtCredentialPayload` type or left for the transformation function to handle.

**Comment by mirceanis** [comment]
Excellent question.
I suppose that answering it would make the spec a lot clearer (and maybe also the implementation).

@awoie do you have an opinion here?

**Comment by joshbax189** [comment]
Also, `normalizeCredential` has the inverse problem:
```
normalizeCredential({
      vc: {
          credentialSubject: { bar: 'baz' },
          '@context': [ 'https://www w3 org/2018/credentials/v1' ],
          type: [ 'VerifiableCredential' ],
          evidence: { foo: 'bar' }
        },
      sub: 'did:ethr:0x435df3eda57154cf8cf7926079881f2912f54db4'
    })
```
Result
```
{
  proof: {},
  vc: { evidence: { foo: 'bar' } },
  credentialSubject: {
    bar: 'baz',
    id: 'did:ethr:0x435df3eda57154cf8cf7926079881f2912f54db4'
  },
  issuer: {},
  type: [ 'VerifiableCredential' ],
  '@context': [ 'https://www.w3.org/2018/credentials/v1' ]
}
```

**Comment by mirceanis** [comment]
yes, and the same for the presentation transforms.

Also it loosely relates to #64 since it relates to JWT-JSON payload conversion.

**Comment by rado0x54** [comment]
Hey, I explicitly fixed the code to handle 'credentialStatus', 'evidence' and 'termsOfUse' correctly. A more general refactor of these conversion functions should be addressed in #64
